frontend/src/utils/api.js

```diff
@@ -51,4 +51,61 @@ function search(searchString) {
   // }).then(res => res.json());
 }
 
-export { register, login, search };
+function getBooksByYearGrade({ year, grade }) {
+  if (year === null && grade === null) {
+    console.log(
+      'getBooksByYearGrade did not receive a year or grade. You should use getBooks instead'
```

Comment:
Could there be any edge case that causes year and grade to be null (i.e. is there FE validation that prevents this method from being called?)

Reply:
This function's parameters are only grabbed from URL params, so probably not haha